### PR TITLE
Add new AD network - Pixels

### PR DESCRIPTION
**Pull request by blademirage**
We have added our ad network "Pixels".
- Add network logic
- Add samples and doc
- Update configuration files

Could you please review our commits? Thank you.

Best regards.

### CODE REVIEW

**Comment by blademirage** [comment]
I signed it!

**Comment by blademirage** [comment]
Signed the CLA using commit author email address.

**Comment by calebcordry** [comment]
Thanks @blademirage. I saw your message on slack and went ahead and restarted the travis build (failure doesn't appear related to your PR. I will review as soon as I have a bit of time.

**Comment by blademirage** [comment]
Thanks @calebcordry.

File: ads/pixels.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+<!---
```
Comment (calebcordry) [inline ads/pixels.md]: Please link to this doc from [here](https://github.com/ampproject/amphtml/blob/master/extensions/amp-ad/amp-ad.md#supported-ad-networks)

File: examples/pixels.amp.html [hunk 2]
```diff
@@ -0,0 +1,45 @@
+<!doctype html>
```
[the remark this answers is not in the document]
Comment (blademirage) [reply examples/pixels.amp.html]: Deleted.

File: ads/pixels.md (hunk 1 above)
Comment (blademirage) [reply ads/pixels.md]: Linked.

File: ads/pixels.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        type="pixels"
+        data-origin="af"
+        data-sid="2847717911664"
+        data-tag="sync"
+        data-click-tracker="false"
+        data-viewability="true"
+>
+</amp-ad>
+<br/>
+<amp-ad width="300" height="250"
```
[the remark this answers is not in the document]
Comment (blademirage) [reply ads/pixels.md]: They are the same type of ads. Removed duplicated example to avoid confusion.

File: examples/ads.amp.html [hunk 4]
```diff
@@ -1199,6 +1200,34 @@ <h2>Outbrain widget</h2>
...
+  <h2>Pixels Examples</h2>
+  <amp-ad width="300" height="250"
+          type="pixels"
+          data-origin="af"
+          data-sid="2847717911664"
+          data-tag="sync"
+          data-click-tracker="false"
+          data-viewability="true">
+  </amp-ad>
+  <br/>
```
Comment (calebcordry) [inline examples/ads.amp.html]: remove these please

File: examples/ads.amp.html [hunk 5]
```diff
@@ -1199,6 +1200,34 @@ <h2>Outbrain widget</h2>
       data-testMode="true">
   </amp-embed>
 
+  <h2>Pixels Examples</h2>
+  <amp-ad width="300" height="250"
+          type="pixels"
```
Comment (calebcordry) [inline examples/ads.amp.html]: can you match indentation of other examples

File: ads/pixels.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+ */
+
+import {validateData, writeScript} from '../3p/3p';
+
+export function pixels(global, data) {
+  validateData(data,
+      ['origin', 'sid', 'tag'],
+      ['clickTracker', 'viewability']
+  );
+  data.tag = (data.tag).toLowerCase();
```
Comment (calebcordry) [inline ads/pixels.js]: this is up to you but any attributes could potentially come through as a number

Comment (blademirage) [reply ads/pixels.js]: Updated.

File: examples/ads.amp.html (hunk 5 above)
Comment (blademirage) [reply examples/ads.amp.html]: Updated.

File: examples/ads.amp.html (hunk 4 above)
Comment (blademirage) [reply examples/ads.amp.html]: Removed.

**Comment by calebcordry** [comment]
Hello @blademirage thanks for making the changes.

We have updated our Percy files since you started this PR. Could you please rebase to upstream/master?

**Comment by blademirage** [comment]
Rebased.

**Comment by calebcordry** [comment]
Hmmm looks like something weird happened with the rebase. Could you try with the instructions [here](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#pull-the-latest-changes-from-the-amphtml-repository)?

**Comment by blademirage** [comment]
I have done a reset on the pixels-amp branch and then rebased to
upstream/master by following the instructions.
Thanks.

**Comment by calebcordry** [comment]
@blademirage Thanks! This is looking good. Is this ready to be merged?

**Comment by blademirage** [comment]
Yes, please merge. Thank you.

**Pull request merged**